### ISSUE TITLE
E3002 Invalid Property Resources/xxx/Properties/FilterPolicyScope. Did you mean FilterPolicy?

### ISSUE THREAD

**Issue by lvthillo** [aws-cloudformation/cfn-lint#2500]
### CloudFormation Lint Version

cfn-lint 0.72.0

### What operating system are you using?

Mac (Intel)

### Describe the bug

`cfn-lint` complains about `FilterPolicyScope`:

```
E3002 Invalid Property Resources/AlertSubscription/Properties/FilterPolicyScope. Did you mean FilterPolicy?`
```

It seems it's also not yet here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-sns-subscription.html

But it's here: https://aws.amazon.com/blogs/compute/introducing-payload-based-message-filtering-for-amazon-sns/

It also works when I try to deploy my subscription with a `FilterPolicyScope`

### Expected behavior

No complains

### Reproduction template

```yaml
  AlertSubscription:
    Type: "AWS::SNS::Subscription"
    Properties:
      Endpoint: "xxx"
      Protocol: "email"
      FilterPolicyScope: MessageBody
      FilterPolicy: ...
```

**Comment by TheConner** [comment]
Coincidentally I ran into this today. It appears that the [schema used by cfn-lint](https://d1uauaxba7bl26.cloudfront.net/latest/gzip/CloudFormationResourceSpecification.json) is missing that property
![image](https://user-images.githubusercontent.com/59540839/203850937-44210fa0-9893-4907-a3a4-f6c5c674ce2c.png)

I submitted a PR to the [cfn-docs](https://github.com/awsdocs/aws-cloudformation-user-guide/pull/1289) to add this attribute in (it was missing there); however, I'm not sure if this will trickle down to the JSON schema.

**Comment by kddejong** [comment]
Do you know if the `FilterPolicyScope` is being set on the subscription?  I want to make sure its not only accepted but actually works.

**Comment by lvthillo** [comment]
Yes, check the example [an official blog](https://aws.amazon.com/blogs/compute/introducing-payload-based-message-filtering-for-amazon-sns/) + I'm also using it in a project already so I confirm this works:
```
AutoInsuranceEventsSubscription:
    Type: AWS::SNS::Subscription
    Properties:
      Protocol: sqs
      Endpoint: !GetAtt AutoInsuranceEventsQueue.Arn
      TopicArn: !Ref InsuranceEventsTopic
      FilterPolicyScope: MessageBody
      FilterPolicy:
        '{"Records":{"s3":{"object":{"key":[{"prefix":"auto-"}]}}
        ,"eventName":[{"prefix":"ObjectCreated:"}]}}'
  
HomeInsuranceEventsSubscription:
    Type: AWS::SNS::Subscription
    Properties:
      Protocol: sqs
      Endpoint: !GetAtt HomeInsuranceEventsQueue.Arn
      TopicArn: !Ref InsuranceEventsTopic
      FilterPolicyScope: MessageBody
      FilterPolicy:
        '{"Records":{"s3":{"object":{"key":[{"prefix":"home-"}]}}
        ,"eventName":[{"prefix":"ObjectCreated:"}]}}'

`
``

**Comment by TheConner** [comment]
From my experience setting `FilterPolicyScope: MessageBody` on the subscription will cause the created subscription to have the filter set to the message body. [here is an example usage from the AWS SNS examples](https://github.com/aws-samples/aws-sns-samples/blob/e2144907411cd3edbb63bcf72f4f52916e0386fc/templates/SNS-Payload-Based-Filtering-SAM.template#L51-L59)

**Comment by lvthillo** [comment]
Hmm then that's working too. But we need to some sort of migration and have a mix between attribute based subscriptions and payload based. So we need the configuration on subscription level.

**Comment by kddejong** [comment]
I see some movement that makes me think that this property will be included shortly in the CloudFormation specs.  I can patch it in if needed.  I'm waiting to see if this weeks spec will include it or not and will add it in necessary.

**Comment by kddejong** [comment]
https://github.com/aws-cloudformation/cfn-lint/releases/tag/v0.72.5
